File: Tutorials/Tuckshop/Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Customers/Commands/DeleteCustomer.cs
namespace Tuckshop.Core.Models.Customers.Commands
{
  using Neo.Model;

  /// <summary>
  /// Class representing a request to delete a customer.
  /// </summary>
  public class DeleteCustomer : ModelBase<DeleteCustomer>
  {
    /// <summary>
    /// Gets and sets the CustomerId.
    /// </summary>
    public int CustomerId { get; set; }
  }
}

File: Tutorials/Tuckshop/Client/Tuckshop.WebApp/src/Domain/Models/Customers/Commands/DeleteCustomer.ts
import { ModelBase, Validation } from '@singularsystems/neo-core';
export default class DeleteCustomer extends ModelBase {
    static typeName = "DeleteCustomer";

    constructor() {
        super();
        this.makeObservable();
    }

    public customerId: number = 0;

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<DeleteCustomer>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        if (this.isNew) {
            return "New delete customer";
        } else {
            return "Delete Customer";
        }
    }
}